frontend/test/pytest/test_mid_circuit_measurement.py

```diff
@@ -210,6 +212,256 @@ def circuit(x):
         assert circuit(0.0) == [0] * 10
         assert circuit(jnp.pi) == [1] * 10
 
+    def test_dynamic_one_shot_with_sample_single(self, backend):
+        """Test that a measurement result can be returned with qml.sample and shots."""
+
+        dev = qml.device(backend, wires=1, shots=10)
+
+        @qjit
+        @dynamic_one_shot
+        @qml.qnode(dev)
+        def circuit(x):
+            qml.RY(x, wires=0)
+            m = measure(0)
+            qml.PauliX(0)
+            return qml.sample(m)
+
+        assert jnp.allclose(circuit(0.0), 0)
+        assert jnp.allclose(circuit(jnp.pi), 1)
+
+    @pytest.mark.parametrize("shots", [8000])
```

Comment:
Generally I would favour smaller shot numbers to keep test times low, but maybe you have wrestle with noise in this case?

Reply:
We are testing `expval` with `atol=0.01, rtol=0.2` and `counts` with `atol=20, rtol=0.2`, so fairly high tolerance and even at 8000 shots tests are a bit flaky. We have to pick our poison :( That said, it would be really nice to come up with a better way to test this functionality. If `jax.jit`ing DQ could be made as fast as Catalyst, we could actually crank up the shots and solve both issues.

Reply:
Oh nice, so 10k shots is super fast with with catalyst on these tiny circuits? 

Not sure how the results are being validated, but couldn't compare against the ideal values, rather than simulating noisily with default.qubit?

Reply:
> Not sure how the results are being validated, but couldn't compare against the ideal values, rather than simulating noisily with default.qubit?

I'm using shots everywhere because of `qml.counts` and `qml.sample`. When possible I use `defer_measurements` (small circuits, few MCMs) so it's quite fast. Otherwise, I have to use `dynamic_one_shot` which makes for a few slow tests, but not too bad I'd say. Conditionally using `shots=None` would help a bit the flakiness, but not that much in my experience.